Magic folder integration tests

meejah opened the pull request:
This ports all the magic-folders integration-style tests (from `check_magicfolder_smoke.py`) to a proper event-runner, etc. Uses `py.test` fixtures to build up the introducer, storage servers and alice + bob clients. It also runs a `flog_tool` gather instance to grab all logs from all the running services and dumps them at the end of the run.

daira commented on integration/conftest.py:
```diff
@@ -0,0 +1,511 @@
+    try:
+        mkdir(join(temp_dir, 'magic-alice'))
+    except OSError:
+        pass
```
Put this before creating the process.

daira commented on integration/conftest.py:
```diff
@@ -0,0 +1,511 @@
+            reactor, request, temp_dir, tahoe_binary, introducer_furl, flog_gatherer, "alice",
+            web_port="tcp:9980:interface=localhost",
+            storage=False,
+#            magic_text='Completed initial Magic Folder scan successfully',
```
This commented-out thing is kind of misleading.

daira commented on integration/conftest.py:
```diff
@@ -0,0 +1,511 @@
+    # enough until the grid is really-ready .. maybe re-try this
+    # command a couple times automagically? or figureout what's up w/
+    # the storage nodes?
+    import time ; time.sleep(5)
```
Eek

daira commented on integration/test_aaa_aardvark.py:
```diff
@@ -0,0 +1,23 @@
+# XXX FIXME just putting these here as a cheap hack at getting some
```
I'm fine with this, remove the XXX FIXME.

meejah commented:
Note to self: try on windows-vm to see if the appveyor failure is "real" or not.

warner commented:
I've kicked appveyor to re-build that one, maybe it was something transient.

meejah commented:
Okay, I've addressed the comments and this is ready to go. Note that two of the tests will fail until the PR for 2814 is landed (if @warner agrees I can squash before merge if you want, perhaps after 2814 is landed so I can rebase it as well).

warner commented:
looks reasonable to me, squash at will

meejah commented:
Okay, I rebased to master and just squashed everything into one commit.
